add `OpInfo` for `torch.repeat_interleave`

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65469
* #65468
* #65467
* #65460
* __->__ #65455
* #65454

Addresses facebookresearch/functorch#103.

Differential Revision: [D31111696](https://our.internmc.facebook.com/intern/diff/D31111696)

pmeier committed:
add `OpInfo` for `torch.repeat_interleave`

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31111696).

zou3519 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3445,6 +3445,16 @@ def sample_inputs_spectral_ops(self, device, dtype, requires_grad=False, **kwarg
+    shape = (2, 3, 4)
+    make_input = partial(make_tensor, shape, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    return [
```
Could we add a scalar_tensor case?
```
x = torch.randn([])
x.repeat_interleave(3)
```

zou3519 reviewed (commented):
This looks pretty good to me. I added a suggestion for a minor test case

pmeier committed:
Update on "add `OpInfo` for `torch.repeat_interleave`"


Addresses facebookresearch/functorch#103.

Differential Revision: [D31111696](https://our.internmc.facebook.com/intern/diff/D31111696)

[ghstack-poisoned]

zou3519 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3445,6 +3445,16 @@ def sample_inputs_spectral_ops(self, device, dtype, requires_grad=False, **kwarg
+    return [
+        SampleInput(make_input(()), kwargs=dict(repeats=2)),
+        SampleInput(make_input((2, 3 ,4)), kwargs=dict(repeats=2)),
+        SampleInput(make_input((2, 3 ,4)), kwargs=dict(repeats=2, dim=1)),
```
Please fix the lint

zou3519 commented:
@pytorchbot ciflow rerun -l ciflow/all

zou3519 reviewed (changes requested):
requesting changes for the lint

pmeier committed:
Update on "add `OpInfo` for `torch.repeat_interleave`"


Addresses facebookresearch/functorch#103.

Differential Revision: [D31111696](https://our.internmc.facebook.com/intern/diff/D31111696)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31111696).
